test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetIVsApiTests/NuGetConsoleTestCase.cs

```diff
@@ -141,10 +135,19 @@ public void UpdatePackageFromPMC()
                 Assert.True(nugetConsole.IsPackageInstalled(packageName, packageVersion2));
 
                 nugetConsole.Clear();
-                solutionService.Close();
             }
         }
 
+        private void OpenNuGetPMC()
+        {
+            var dte = VisualStudio.Dte;
+            dte.ExecuteCommand("View.PackageManagerConsole");
+            // We need to wait for the tool window to be initialized
+            // Because it is started on idle there is no way to make sure it
+            // has been initialized, so just pause for 10 seconds.
+            Thread.Sleep(10000);
```

Comment:
This is okay as a temporary fix, but we really need a way to avoid all sleeps in tests.  These cause so many problems.

Reply:
Is it possible to continually query the output window looking for some text instead until it appears?

Reply:
It could look for `PM>` to detect when the window is loaded. Long term we should add an internal event or property we can check.

Reply:
@jainaashish do you think there could be a way to make sure PMC has started instead of sleeping?

Reply:
@jainaashish any suggestion?

Reply:
You can look at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L283 here we try to look for the already exist tool window, if not then create it similarly you can also do something just dont create if it doesn't exist.